sdk/core/core-lro/src/http/models.ts

```diff
@@ -110,3 +110,8 @@ export interface CreateHttpPollerOptions<TResult, TState> {
    */
   resolveOnUnsuccessful?: boolean;
 }
+
+export interface LongRunningOperationLocationHeaders {
```

Comment:
What is the reason to add and export this model?

Reply:
There's a task https://github.com/Azure/autorest.typescript/issues/1452 in cadl support to add the basic types from cadl-azure-core into each language's core libraries.

Reply:
I see, has the `LongRunningOperationLocationHeaders` name been reviewed by the architecture board? Also, could you share an example of the generated code that uses that type?

Reply:
Not yet, I just checked with @joheredi. 
Here's an example of the generated code.
LongRunningOperationLocationHeaders https://github.com/Azure/autorest.typescript/blob/068ef43c49fa949bddc32993b1d41c012d6a2a70/packages/cadl-rlc-test/test/authoring/cadl-output/src/responses.ts
Paged. https://github.com/Azure/autorest.typescript/blob/068ef43c49fa949bddc32993b1d41c012d6a2a70/packages/cadl-rlc-test/test/authoring/cadl-output/src/outputModels.ts

You can learn more by checking this PR https://github.com/Azure/autorest.typescript/pull/1665/files

Reply:
I see. So do you plan on adding a new interface for each header name (e.g. "location" and the azure async operation one)?

Reply:
I think so. Actually, I checked with cadl team, for data plane, operation-location is the recommanded way for LRO. location and azure-async header are for mgmt plane LROs.

Reply:
I don't think we should just add interfaces if we don't have a compelling use case for it. I'm not sure there is value in this kind of interface which just describes a header name. Can you say more about how this would get used in clients? Keep in mind that TS is structural rather than nominal so I think the necessity of interfaces in core libraries is somewhat reduced. A good argument is to ensure things are consistent, but in this case our hands are tied by the header name I think...

Reply:
The reason I want to put the lro header into the core-lro library is that I found there will be multiple headers generated as long as there're LROs defined in the cadl and "operation-location" is the only header defined in the response, like 
```
export interface CreateOrUpdate200Headers {
  /** The location for monitoring the operation state. */
  "operation-location": string;
}
export interface CreateOrUpdate201Headers {
  /** The location for monitoring the operation state. */
  "operation-location": string;
}
export interface Delete202Headers {
  /** The location for monitoring the operation state. */
  "operation-location": string;
}
export interface Export202Headers {
  /** The location for monitoring the operation state. */
  "operation-location": string;
}
```
The structure is all the same and the only difference is the header type name which we have to give them an unique name with an operation name + response code prefix. But if we have a type defined in the core-lro, we don't need to generate those duplicate code, just need to import one header type and then reuse it in all the response definition like
```
/** The request has succeeded. */
export interface CreateOrUpdate200Response extends HttpResponse {
  status: "200";
  body: ProjectOutput;
  headers: RawHttpHeaders & LongRunningOperationLocationHeaders;
}
export interface Delete202Response extends HttpResponse {
  status: "202";
  body: OperationStatusOutput;
  headers: RawHttpHeaders & LongRunningOperationLocationHeaders;
}
```

Reply:
Ok I'm fine with this if we think people will use it.